fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/OnlineIndexer.java

```diff
@@ -560,6 +566,11 @@ private void increaseLimit() {
             if (recordsScanned != null) {
                 recordsScanned.incrementAndGet();
             }
+            if (trackProgress) {
+                // Add one to the number of records successfully scanned and processed.
+                store.context.ensureActive().mutate(
+                        MutationType.ADD, scannedRecordsSubspace.getKey(), FDBRecordStore.LITTLE_ENDIAN_INT64_ONE);
```

Comment:
@alecgrieser did not raise this as an issue, so I doubt it is one, but would there be any benefit to doing this mutation only once, just before the commit of the scanned range, rather than doing it on each record scanned?  For example, in the proposed RPC interface to FDB (https://github.com/apple/foundationdb/wiki/FoundationDB-RPC-Layer-Requirements) this operation would, presumably, add a round trip each time.

Reply:
Hm. With the current client implementation, the only benefit to doing it all at once at the end would be that there would be approximately one fewer JNI hop and a little bit of CPU work on the client, but because the FDB client coalesces mutations, it wouldn't end up issuing one mutation per record here, it would just be one single mutation at the end (similar to if the coalescing were done at the Record Layer level).

I think the plan (or hope) would be that an RPC layer would have some kind of client that also batched updates together in some way (so that, for example, multiple mutations wound up on the same packet to the RPC server), though the overhead would be a little bit higher than if there was just one mutation (for example, you would need to repeat the key multiple times). But that's also somewhat speculative, on my part, given that I think that's still a little up in the air.

So, in conclusion, probably would be a non-zero savings to only have mutation, but also probably not all that expensive the way it is.

Reply:
Collapsed the counts, considering it won't hurt anyways. (I didn't do it initially mainly because I didn't want to write things to encode a `long`)